intel-sgx/dcap-artifact-retrieval/src/provisioning_client/mod.rs

```diff
@@ -534,10 +534,12 @@ pub trait ProvisioningClient {
 
     fn qe_identity(&self) -> Result<QeIdentitySigned, Error>;
 
-    /// Retrieve PCK certificates using `pckcerts()` and fallback to `pckcert()`
-    /// if provisioning client does not support pckcerts. Note that in case of
-    /// fallback the returned PckCerts will only contain a single certificate
-    /// associated with the highest TCB level applied to the platform.
+    /// Retrieve PCK certificates using `pckcerts()` and fallback to the
+    /// following method if that's not supported:
+    /// - Call `pckcert()` to find the FMSPC.
+    /// - Using the FMSPC value, call `tcbinfo()` to get TCB info.
+    /// - For each TCB level in the result of previous call:
+    ///   - Call `pckcert()` to get the PCK cert associated with that TCB level.
     fn pckcerts_with_fallback(&self, pck_id: &PckID) -> Result<PckCerts, Error> {
```

Comment:
Do we still need a special function for that? Can't we just implement the fallback when it's needed?

Reply:
This is that implementation, whether or not it's used is up to the caller. I think we have two places we need to call this, that's why I have implemented it here to avoid duplication.

Reply:
Oh you can't just always use the `pckcerts` function, because you need more info.